### PR TITLE
Documentation rework

### PR DESCRIPTION
**Pull request by kacperkapusciak**
## Description

This PR adds new pages with screenshots and videos showcasing the odds and ends of the Radon IDE extension.

Also, it changes the structure of the documentation by splitting the existing documentation into 3 sections:  

- `Getting started` describing installation, UI and features overview,
- `Features` goes in depth about the all the features provided by the extension,
- `Guides` hosts guides that might be useful while using Radon IDE.

## Added or overhauled pages
- [x] Device management
- [x] User interface
- [x] Debugging and logging 
- [x] Element inspector
- [x] Device settings
- [x] Router integration
- [x] Previews
- [x] Keyboard shortcuts
- [x] Screen recording
- [x] Storybook integration

## Missing pages

- [ ] Touch and multitouch

## Testing

Visit the documentation preview provided by Vercel on this PR

Alternatively you can run the documentation from my branch:

```
cd packages/docs
npm i
npm run start
```

### CODE REVIEW

**Comment by maciekstosio** [comment]
Have you tested it on different browsers? 

|Chrome|Safari|
|-|-|
|SCR-20241105-iqtm|SCR-20241105-iqqs|

**Comment by maciekstosio** [comment]
What do you think about adding a forward feature to the main page? This is a pretty neat feature that I haven't seen anywhere else.

**Comment by maciekstosio** [comment]
I'm not good at lawyerish, but this sounds like you can distribute your copies on our channels :D 
image

**Comment by maciekstosio** [comment]
image
href mailto ?

File: packages/docs/docs/features/previews.mdx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+<Tabs groupId="package-managers">
+  <TabItem value="npm" label="NPM">
+
+    npm i radon-ide
+
+  </TabItem>
+  <TabItem value="yarn" label="YARN">
+
+    yarn add radon-ide
```
Comment (maciekstosio) [inline packages/docs/docs/features/previews.mdx:22]: Shouldn't it be a dev dependency?

**Comment by maciekstosio** [comment]
image
Caution box?

**Comment by maciekstosio** [comment]
image
Busted

**Review by maciekstosio — CHANGES_REQUESTED** [review]
My suggestions. And it's small thing but we use bunch of forms for VSCode "vscode" "VS Code" "VSCode" maybe it would be good to unify that.

File: packages/docs/docs/guides/troubleshooting.md [hunk 2]
```diff
@@ -17,21 +17,21 @@ In this case lookup command called "Radon IDE: Diagnostics" – when executed i
...
+If your project doesn't work out of the box because of some modifications made to its setup, chances are the Radon IDE can be configured to support your modifications, please check [configuration](/docs/getting-started/launching) guide to learn more.
 
 ### -sec-num- Can extension locate your React Native project
 
 The extension supports monorepo-type of setups and you should be able to use it even if the app isn't in the root folder of your workspace.
 This setup however has some limitations because the extension can only work with a single application per workspace.
 If your monorepo contain multiple projects, you'll need to instruct the extension with the location of your main React Native or Expo application folder.
-This can be done using [launch configuration](./launch-configuration.md) using `appRoot` setting – please refer to [configuration guide](./launch-configuration.md) for more details.
+This can be done using [launch configuration](/docs/getting-started/launching) using `appRoot` setting – please refer to [configuration guide](/docs/getting-started/launching) for more details.
```
Comment (maciekstosio) [inline packages/docs/docs/guides/troubleshooting.md:28]: Shouldn't all of them point to `guides/configuration`?

File: packages/docs/docs/guides/troubleshooting.md [hunk 3]
```diff
@@ -17,21 +17,21 @@ In this case lookup command called "Radon IDE: Diagnostics" – when executed i
...
 
 ### -sec-num- List of available devices is empty
 
 The extension relies on Android Studio to install Android emulators and on Xcode tools to manage and install iOS simulators.
 The extension can spawn new devices on its own, but it requires the system images can only be installed using Android Studio and Xcode.
```
Comment (maciekstosio) [inline packages/docs/docs/guides/troubleshooting.md:33]:
```suggestion
The extension can spawn new devices on its own, but it requires the system images to be installed using Android Studio and Xcode.
```

**Review by maciekstosio — CHANGES_REQUESTED** [review]
Few more

File: packages/docs/docs/getting-started/feature-highlight.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Learn more about the [Debugging and logging](/docs/features/debugging-and-logging).
+
+<video autoPlay loop width="700" controls className="shadow-image">
+  <source src="/video/5_sztudio_logs_panel.mp4" type="video/mp4"/>
+</video>
+
+## Router integration
+
+The Radon IDE integrates with your deep-linked application allowing you to jump around the navigation structure.
```
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply packages/docs/docs/getting-started/feature-highlight.md:44]: That's the industry name for having an url for some screens in your mobile app. Your users can click a link on a website that navigates them to _deeply nested_ screen in your app. 

https://reactnavigation.org/docs/deep-linking/
https://developer.android.com/training/app-links/deep-linking

File: packages/docs/docs/getting-started/user-interface.mdx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+## Overview
+
+After [launching](/docs/getting-started/launching) the the Radon IDE panel the user interface should look as follows:
+
+<img width="600" src="/img/docs/ide_user_interface.png" className="shadow-image" />
+
+1. **Back button** - goes to the previous URL route in the navigation history.
+2. **Reload button** - reloads the JavaScript bundle from Metro bundler. Hovering over the reload button reveals an arrow which allows for more granular reload options like:
+
+   - _Reload JS_ - reloads the JavaScript bundle from Metro bundler,
```
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply packages/docs/docs/getting-started/user-interface.mdx:16]: yeah, maybe let's not do this 😅 
was-it-even-worth-v0-n0dugfgrz5l81

File: packages/docs/docs/getting-started/user-interface.mdx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+7. **Zoom controls** - hovering over the area reveals the zooms controls which allow to adjust how zoomed the device preview should be. You can choose one of the predefined values or use the plus and minus magnifier buttons to adjust the zoom manually.
+   <img width="400" src="/img/docs/ide_zoom_controls.png" />
+8. **Device preview** - the preview of the running iOS simulator or Android emulator.
+9. [**Element inspector**](/docs/features/element-inspector) - enabling element inspector and clicking on the preview opens the line of code where the highlighted component is defined. Right-clicking on the device preview allows you to select components directly from the React tree.
+10. [**Manage devices**](/docs/guides/device-management) - allows to create, delete, and select devices currently used in the Radon IDE preview.
+    <img width="250" src="/img/docs/ide_manage_devices.png" />
+11. **Report issue** - opens a page in the Radon IDE GitHub repository where you can submit issue reports and suggest features.
+12. [**Device settings**](/docs/features/device-settings) - allows to adjust the device settings like device appearance, location, permissions, and text size.
```
[the remark this answers is not in the document]
Comment (kacperkapusciak) [reply packages/docs/docs/getting-started/user-interface.mdx:32]: i agree! it feels like it should be in the `settings` in the top-right corner of the panel

File: packages/docs/docs/guides/troubleshooting.md (hunk 2 above)
Comment (kacperkapusciak) [reply packages/docs/docs/guides/troubleshooting.md]: You're absolutely right! Fixed in [2c71bd5](https://github.com/software-mansion/radon-ide/pull/689/commits/2c71bd5edec9228cd73f382fd989ddea04362fa3)

File: packages/docs/docs/features/previews.mdx (hunk 1 above)
Comment (kacperkapusciak) [reply packages/docs/docs/features/previews.mdx:24]: I'm pretty sure it needs to be a dependency 🤔

File: packages/docs/docs/features/storybook-integration.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+---
+id: storybook-integration
+title: Storybook integration
+sidebar_position: 7
+---
+
+Radon IDE seamlessly integrates with Storybook. The extension automatically detects Storybook stories and
```
Comment (maciekstosio) [inline packages/docs/docs/features/storybook-integration.md:7]: and?

**Review by maciekstosio — CHANGES_REQUESTED** [review]

File: packages/docs/docs/features/storybook-integration.md (hunk 7 above)
Comment (kacperkapusciak) [reply packages/docs/docs/features/storybook-integration.md]: fixed in [ee9c612](https://github.com/software-mansion/radon-ide/pull/689/commits/ee9c612f824d826cdb220e9d5ad545542ba1f265)

**Review by maciekstosio — APPROVED** [review]
lgtm

**Pull request merged**